packages/headless/doc-parser/src/interface-resolver.ts

```diff
@@ -107,15 +107,16 @@ function resolvePropertySignature(
   ancestorNames: string[]
 ) {
   const typeExcerpt = p.propertyTypeExcerpt.spannedTokens[0];
+  const typeName = extractTypeName(p.propertyTypeExcerpt);
   if (isRecordType(typeExcerpt)) {
     return buildEntityFromProperty(p);
   }
 
-  if (isTypeAlias(typeExcerpt)) {
+  if (isTypeAlias(typeExcerpt) && !typeName.includes('|')) {
```

Comment:
Personally, I would prefer introducing a `isUnionType` function and adding a fourth `if` just above this one that calls the new function.